### PR TITLE
refactor: refactor encoder models and add pytests

### PR DESCRIPTION
**Pull request by rebel-eunji**
### 🚀 Summary of Changes

Encoder models support multiple pooling types, but it was previously hard-coded to `CLS`. This PR:

- Selects the appropriate pooling type per model instead of forcing `CLS`
- Adds pytests to verify pooling behavior
- Drops the no-longer-valid `score` task from the classifier pooler (removed in vLLM 0.22.0; scoring now maps to `classify`)
- Apply activation to logits instead of using raw logits

---

### 📌 Related Issues / Tickets

* Resolves #
* Related to #

---

### ✅ Type of Change

* [ ] 🚀 Release (`release`)
* [ ] ✨ Feature (`feature`)
* [x] 🧠 Model support (`model`)
* [ ] 🧬 Core engine changes (`core`)
* [ ] 🛠 Bug fix (`fix`)
* [ ] ⚙️ Performance improvement (`perf`)
* [ ] 🔁 Refactor or code cleanup (`refactor`)
* [ ] 📄 Documentation (`docs`)
* [ ] ❓ Other (`other`): please describe

---

### 🧪 How to Test

```bash
pytest tests/v1/models/optimum/test_encoder_pooling.py
```

---

### 📋 Checklist

* [x] PR title follows Conventional Commits format
* [ ] This PR is linked to an existing issue
* [x] The test method is described, and the expected result is clearly stated
* [ ] Relevant documentation has been updated (if applicable)

### CODE REVIEW

**Comment by rebel-eunji** [comment]
https://github.com/RBLN-SW/vllm-rbln/pull/644 is applied here. @rebel-thkim

**Review by rebel-jonghewk — APPROVED** [review]
